### PR TITLE
fixed DIG bugs for GS : 1814, 2684, 2685, 2637, 522

### PR DESCRIPTION
**Pull request by minsunny**
https://github.com/nanoPayinc/NANOPAY/issues/1814
https://github.com/nanoPayinc/NANOPAY/issues/2684
https://github.com/nanoPayinc/NANOPAY/issues/2685
https://github.com/nanoPayinc/NANOPAY/issues/2637
https://github.com/nanoPayinc/NANOPAY/issues/522

fixed Putting error, casting error, formatting error above tickets

### CODE REVIEW

File: src/foam/nanos/http/HttpParametersWebAgent.java [hunk 1]
```diff
@@ -122,7 +128,7 @@ public void execute(X x) {
     }
 
     if ( ! "application/x-www-form-urlencoded".equals(contentType) ) {
-      switch ( methodName.toUpperCase() ) {
```
Comment (jlhughes) [inline src/foam/nanos/http/HttpParametersWebAgent.java]: Did you test this with curl or an external POST operation?

**Comment by kgrgreer** [comment]
Looks good to me, but answer Joel's outstanding question.

File: src/foam/nanos/http/HttpParametersWebAgent.java [hunk 2]
```diff
@@ -84,6 +87,9 @@ public void execute(X x) {
     // Capture 'data' on all requests
     if ( ! SafetyUtil.isEmpty(req.getParameter("data")) ) {
       logger.debug("data", req.getParameter("data"));
```
Comment (jlhughes) [inline src/foam/nanos/http/HttpParametersWebAgent.java:89]: now that we have things working, let's cleanup some of the debugging.

Comment (jlhughes) [inline src/foam/nanos/http/HttpParametersWebAgent.java:89]: now that we have things working, let's cleanup some of the debugging.  Just comment them out for now.

File: src/foam/nanos/dig/DigWebAgent.java [hunk 3]
```diff
@@ -196,6 +199,8 @@ public void execute(X x) {
             obj = (FObject)i.next();
             obj = dao.put(obj);
           }
+
+          //returnMessage = "<objects>" + success + "</objects>";
```
Comment (jlhughes) [inline src/foam/nanos/dig/DigWebAgent.java:203]: delete please.

**Review by jlhughes — APPROVED** [review]
Couple of comments to address then ready to go.

File: src/foam/nanos/http/HttpParametersWebAgent.java (hunk 2 above)
Comment (jlhughes) [inline src/foam/nanos/http/HttpParametersWebAgent.java:89]: comment out these debug now that things are working.

File: src/foam/nanos/dig/DigWebAgent.java (hunk 3 above)
Comment (jlhughes) [inline src/foam/nanos/dig/DigWebAgent.java:203]: delete.

Comment (jlhughes) [inline src/foam/nanos/dig/DigWebAgent.java:203]: delete.

**Review by jlhughes — APPROVED** [review]
Couple of comments to address then ready to go.

**Pull request merged**